src/Components/Users/ManageUsers.tsx

```diff
@@ -79,59 +69,50 @@ export default function ManageUsers() {
   const isExtremeSmallScreen =
     width <= extremeSmallScreenBreakpoint ? true : false;
 
-  const fetchData = useCallback(
-    async (status: statusType) => {
-      setIsLoading(true);
-      const params = {
-        limit: resultsPerPage,
-        offset: (qParams.page ? qParams.page - 1 : 0) * resultsPerPage,
-        username: qParams.username,
-        first_name: qParams.first_name,
-        last_name: qParams.last_name,
-        phone_number: qParams.phone_number,
-        alt_phone_number: qParams.alt_phone_number,
-        user_type: qParams.user_type,
-        district_id: qParams.district_id,
-      };
-      if (qParams.district_id) {
-        const dis = await dispatch(getDistrict(qParams.district_id));
-        if (!status.aborted) {
-          if (dis && dis.data) {
-            setDistrictName(dis.data.name);
-          }
-        }
-      } else {
-        setDistrictName(undefined);
-      }
-      const res = await dispatch(getUserList(params));
-      if (!status.aborted) {
-        if (res && res.data) {
-          setUsers(res.data.results);
-          setTotalCount(res.data.count);
-        }
-        setIsLoading(false);
-      }
+  const {
+    data: userListData,
+    loading: userListLoading,
+    refetch: refetchUserList,
+  } = useQuery(routes.userList, {
+    query: {
+      limit: resultsPerPage.toString(),
+      offset: (
+        (qParams.page ? qParams.page - 1 : 0) * resultsPerPage
+      ).toString(),
+      username: qParams.username,
+      first_name: qParams.first_name,
+      last_name: qParams.last_name,
+      phone_number: qParams.phone_number,
+      alt_phone_number: qParams.alt_phone_number,
+      user_type: qParams.user_type,
+      district_id: qParams.district_id,
     },
-    [
-      resultsPerPage,
-      qParams.page,
-      qParams.username,
-      qParams.first_name,
-      qParams.last_name,
-      qParams.phone_number,
-      qParams.alt_phone_number,
-      qParams.user_type,
-      qParams.district_id,
-      dispatch,
-    ]
-  );
+  });
 
-  useAbortableEffect(
-    (status: statusType) => {
-      fetchData(status);
-    },
-    [fetchData]
-  );
+  const fetchDistrict = useCallback(async () => {
+    setIsLoading(true);
+    if (qParams.district_id) {
+      const { data: districtData } = await request(routes.getDistrict, {
+        pathParams: { id: qParams.district_id },
+      });
+      if (districtData) setDistrictName(districtData.name);
+      else setDistrictName(undefined);
+    }
+    setIsLoading(false);
+  }, [qParams.district_id]);
```

Comment:
Replace request being wrapped inside useCallback altogether with useQuery and set the `prefetch` option to enable/disable query on mount.

Remove `isLoading` state in favour of loading state maintained by useQuery